src/engine/renderer/tr_image.cpp

```diff
@@ -2076,6 +2107,11 @@ image_t        *R_FindCubeImage( const char *imageName, int bits, filterType_t f
 	}
 	numPicsToFree = 0;
 
+	// no need to store greaterHeight since cubmap face is a square
```

Comment:
I don't see anything forcing the image to be a square. `R_CreateCubeImage` takes both width and height arguments.

Reply:
The code returns earlier if `width ≠ height`:

https://github.com/DaemonEngine/Daemon/blob/7774ff80c78f2083e23027c7a24561f07b64cf9f/src/engine/renderer/tr_image.cpp#L2092-L2096

And in anyway, a cube is a regular hexahedron, which means all faces are equal squares. :wink: 

It's possible to also fix any potential non-cube hexahedron, btw. reading this code with all those goto and stuff just make me thing the whole can be rewritten…